internal/statesync/metrics.go

```diff
@@ -0,0 +1,91 @@
+package statesync
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/discard"
+	"github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// MetricsSubsystem is a subsystem shared by all metrics exposed by this package.
+	MetricsSubsystem = "statesync"
+)
+
+// Metrics contains metrics exposed by this package.
+type Metrics struct {
+	TotalSnapshots     metrics.Counter
+	ChunkProcess       metrics.Gauge
+	SnapshotHeight     metrics.Gauge
+	SnapshotChunk      metrics.Counter
+	SnapshotChunkTotal metrics.Gauge
+	BackFill           metrics.Counter
```

Comment:
As above, it's not clear what these fields mean. Could the ones that aren't obvious please get brief line-ending comments as guideposts?

From reading the initializations below, I think this maybe is intended to be a flag? (Ideally the reader of this type shouldn't need to go searching for that, though)